bids/variables/io.py

```diff
@@ -345,8 +345,12 @@ def _load_tsv_variables(layout, type_, dataset=None, columns=None,
         # Filter rows on all selectors
         comm_cols = list(set(_data.columns) & set(selectors.keys()))
         for col in comm_cols:
-            vals = listify(selectors.get(col))
-            _data = _data.query('%s in @vals' % col)
+            ix = [False] * _data.shape[0]
+            for val in listify(selectors.get(col)):
+                if not layout.regex_search:
+                    val = "^{}$".format(val)
+                ix = ix | _data[col].str.contains(val)
+            _data = _data[ix]
```

Comment:
Given that this apparently takes regexes, maybe let's be as consistent as possible with [grabbit.File._matches()](https://github.com/grabbles/grabbit/blob/5a588731d1a4a42a6b67f09ede110d7770845ed0/grabbit/core.py#L77-L88):

```Python
def make_patt(x):
    patt = '%s' % x
    if isinstance(x, (int, float)):
        # allow for leading zeros if a number was specified
        # regardless of regex_search
        patt = '0*' + patt
    if not regex_search:
        patt = '^%s$' % patt
    return patt

ent_patts = [make_patt(x) for x in listify(selectors.get(col))]
patt = '|'.join(ent_patts)

_data = _data[_data[col].str.contains(patt)]
```

Would probably make sense in the next version to go ahead and abstract out the `patt` generation sequence into a reusable utility function.

Hopefully there's some subtle bug in the current implementation that this will resolve, too, because I have no idea what's causing the issue I posted.